src/coreclr/tools/Common/JitInterface/CorInfoInstructionSet.cs

```diff
@@ -768,6 +768,59 @@ public InstructionSetInfo(string name, string managedName, InstructionSet instru
             }
         }
 
+        public static IEnumerable<string> AllCpuFamilies()
+        {
+            // Only report these in the help, others are added to CpuFamilyToInstructionSets just for muscle memory after clang/llvm
+            yield return "generic";
+            yield return "nehalem";
+            yield return "sandybridge";
+            yield return "haswell";
+            yield return "ampere-altra";
+            yield return "apple-m1";
+        }
+
+        public static IEnumerable<string> CpuFamilyToInstructionSets(string cpu, TargetArchitecture arch)
+        {
+            string sets = null;
+
+            if (arch == TargetArchitecture.ARM64)
+            {
+                sets = cpu switch
+                {
+                    "generic" => "neon",
+
+                    "ampere-altra" => "neon,lse",
+
+                    "apple-m1" => "neon,aes,crc,dotprod,rdma,sha1,sha2,lse,rcpc",
+
+                    _ => null
+                };
+            }
+            else if (arch == TargetArchitecture.X64 || arch == TargetArchitecture.X86)
+            {
+                sets = cpu switch
+                {
+                    "generic" => "sse2",
+
+                    "nehalem" => "sse4.2",
+
+                    // Rosetta2
+                    "apple-m1" => "sse4.2,pclmul",
```

Comment:
The set is a function of the macOS version that Rosetta2 is part of, not a function of having apple-m1 chip.